File: Piko/Sources/SFBuffer.m
//
//  SFBuffer.m
//
//  Copyright (c) 2014 Satoshi Muraki. All rights reserved.
//

#import "SFBuffer.h"

#import "SFDebug.h"

@implementation SFBuffer

- (instancetype)initWithNumberOfFrames:(NSInteger)frames
{
    self = [super init];
    if (self != nil) {
        SFAssert(0 < frames);
        _numberOfFrames = frames;
        _data = malloc(sizeof(double) * frames);
        SFAssert(_data != NULL);
    }
    return self;
}

- (void)dealloc
{
    free(_data);
}

@end

File: Piko/Sources/SFBuffer.swift
//
//  SFBuffer.swift
//
//  Copyright (c) 2014 Satoshi Muraki. All rights reserved.
//

import Foundation

@objc class SFBuffer: NSObject {

    let numberOfFrames: Int
    let data: UnsafeMutablePointer<Double>

    init(numberOfFrames: Int) {
        self.numberOfFrames = numberOfFrames
        data = UnsafeMutablePointer<Double>.alloc(numberOfFrames)
    }

    deinit {
        data.dealloc(numberOfFrames)
    }
}
